### PR TITLE
Twitter Integration

### PR DESCRIPTION
**Pull request by arminm**
#12 Twitter Integration

API key and the secret key should be changed since they are exposed here. Couldn't think of a practical way to share these keys for an Android project.

### CODE REVIEW

**Comment by dsarlis** [comment]
Hmmmm, what I always follow as a golden rule is not to push such keys in git. How do you get this key? I'm guessing you get it after the user is authenticated in twitter, right? If so, this would be done dynamically so we won't even have that info the moment that the app is installed. For now it is ok I guess, if we just use it for testing purposes.

**Comment by arminm** [comment]
@dsarlis These are API keys, assigned to our app on Twitter. They won't change.

**Comment by arminm** [comment]
We could pass them in from the backend in the login response.

**Comment by dsarlis** [comment]
I think this is a better choice. But since these are exposed, we should do that with a new set of keys.

**Comment by arminm** [comment]
Yeah we can regenerate them easily from the dashboard on twitter.

**Comment by dsarlis** [comment]
:+1:

File: CaughtUp/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,6 +18,10 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <meta-data
+            android:name="io.fabric.ApiKey"
+            android:value="a0y2v6pCXfyvHgf6HUU8KPdaJ" />
     </application>
 
+    <uses-permission android:name="android.permission.INTERNET" />
```
Comment (dsarlis) [inline CaughtUp/app/src/main/AndroidManifest.xml:45]: I guess this is not enough for API 23. We have to dynamically request this permission when the user first tries to use the app.

Comment (arminm) [inline CaughtUp/app/src/main/AndroidManifest.xml:45]: This was part of the Twitter docs to add so I would leave it in there just in case.

**Comment by dsarlis** [comment]
How about changing this, so it follows the same approach as the facebook integration using the keys.xml file?

File: CaughtUp/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (valesken) [inline CaughtUp/app/src/main/AndroidManifest.xml:45]: Agreed, we should leave it. It's required even for API 23, it just might not be enough. Necessary but not sufficient, etc.

**Pull request merged**